include/cuda/std/detail/libcxx/include/support/atomic/atomic_cuda_derived.h

```diff
@@ -8,8 +8,12 @@
 //
 //===----------------------------------------------------------------------===//
 
+#include "atomic_cuda_local.h"
+
 template<class _Type, class _Scope, typename _CUDA_VSTD::enable_if<sizeof(_Type) <= 2, int>::type = 0>
 bool __device__ __atomic_compare_exchange_cuda(_Type volatile *__ptr, _Type *__expected, const _Type *__desired, bool, int __success_memorder, int __failure_memorder, _Scope __s) {
+    bool __ret;
+    if (__cuda_compare_exchange_weak_if_local(__ptr, __expected, __desired, &__ret)) return __ret;
```

Comment:
I assume this doesn't add overhead because the `__isLocal` check is optimized at compile time?

Reply:
It will add overhead if it isn't optimized, but without it, you'll get wrong results faster.

There is a better patch we can do later were we recover this overhead, but that will require some more changes throughout the stack.